Added hints to three problems (#3477)
cpinitiative/usaco-guide: solutions/gold/usaco-1211.mdx

[Official Analysis (Java)](http://usaco.org/current/data/sol_prob3_gold_feb22.html)

<Spoiler title="Hint 1">

> Maybe we can somehow exploit the y constraint $0 \le y_i \le 10$ given to each cow?

Reviewer: ```suggestion
Notice that $0 \le y_i \le 10$. There must be a way to exploit this constraint somehow.
```

</Spoiler>

<Spoiler title="Hint 2">

> Maybe there exists a workaround that can allow us to establish less edges at beginning while ensuring all the edges that should belong to the MST are not filtered out?

Reviewer: replace "Maybe there exists" with uh "Try to find"
so in this case
```suggestion
Try to find a workaround that can allow us to establish less edges at beginning while ensuring all the edges that should belong to the MST are not filtered out.
```

</Spoiler>

<Spoiler title="Hint 3">

> Given the coordinates of three cows. Can we somehow use the triangle inequality theorem to our advantage to filter out an edge?

Reviewer: ```suggestion
Could we use the [Triangle Inequality](https://en.wikipedia.org/wiki/Triangle_inequality) somehow?
```

[Official Analysis (Java)](http://usaco.org/current/data/sol_prob3_gold_feb22.html)

<Spoiler title="Hint 1">

We can figure out this problem require the [Kruskal's MST algorithm](https://usaco.guide/gold/mst#kruskals) in some way. We know this because it asks to construct a minimum spanning tree among the cows, and a minimum spanning tree uses the shortest edges to connect all nodes.

</Spoiler>

<Spoiler title="Hint 2">

We soon find out that to pick out the best edges for the minimum spanning tree, we first have to establish a fully connected graph containg $N^2$ edges. 

However, there exists a workaround that can allow us to establish less edges at beginning while ensuring all the edges that should belong to the MST are not filtered out. 

> Maybe we can somehow exploit the y constraint $0 \le y_i \le 10$ given to each cow?

Reviewer: split this into some smaller hints ig

<Spoiler title="Hint 2">

Maybe there exists a workaround that can allow us to establish less edges at beginning while ensuring all the edges that should belong to the MST are not filtered out?

</Spoiler>

<Spoiler title="Hint 3">

Maybe we can somehow exploit the y constraint $0 \le y_i \le 10$ given to each cow?

</Spoiler>

<Spoiler title="Hint 4">

> Given the coordinates of three cows. Can we somehow use the triangle inequality theorem to our advantage to filter out an edge?

Reviewer: this hint seems like it should come earlier tbh

<Spoiler title="Hint 3">

Given the coordinates of three cows. Can we somehow use the triangle inequality theorem to our advantage to filter out an edge?

</Spoiler>

<Spoiler title="Hint 4">

> Maybe we can somehow exploit the y constraint $0 \le y_i \le 10$ given to each cow?

Reviewer: this should be the first one imo

Author: like this?

MST > exploit y constraint > explain how the workaround work > triangle inequality

Author: oh ok
